src/main/java/com/minecolonies/coremod/entity/citizen/EntityCitizen.java

```diff
@@ -399,6 +414,195 @@ public InteractionResult checkAndHandleImportantInteractions(final Player player
+        {
+            if (isBaby())
+            {
+                childFoodInteraction(usedStack, player, hand);
+            }
+            else
+            {
+                eatFoodInteraction(usedStack, player, hand);
+            }
+            return InteractionResult.CONSUME;
+        }
+
+        if (usedStack.getItem() == Items.BOOK && isBaby())
+        {
+            usedStack.setCount(usedStack.getCount() - 1);
+            player.setItemInHand(hand, usedStack);
+
+            if (!level.isClientSide())
+            {
+                getCitizenData().getCitizenSkillHandler().addXpToSkill(Skill.Intelligence, 50, getCitizenData());
+            }
+
+            interactionCooldown = 20 * 60 * 5;
+            return InteractionResult.CONSUME;
+        }
+
+        if (usedStack.getItem() == Items.CACTUS)
+        {
+            usedStack.setCount(usedStack.getCount() - 1);
+            player.setItemInHand(hand, usedStack);
+
+            if (!level.isClientSide())
+            {
+                player.sendMessage(new TranslatableComponent("com.minecolonies.coremod.interaction.ouch", getCitizenData().getName()), player.getUUID());
+                getNavigation().moveAwayFromLivingEntity(player, 5, 1);
+            }
+
+            interactionCooldown = 20 * 60 * 5;
+            return InteractionResult.CONSUME;
+        }
+
+        if (usedStack.getItem() == Items.GLOWSTONE_DUST)
+        {
+            usedStack.setCount(usedStack.getCount() - 1);
+            player.setItemInHand(hand, usedStack);
+
+            if (!level.isClientSide())
+            {
+                addEffect(new MobEffectInstance(MobEffects.GLOWING, 20 * 60 * 3));
+            }
+
+            interactionCooldown = 20 * 60 * 3;
+            return InteractionResult.CONSUME;
+        }
+
+        return null;
+    }
+
+    /**
+     * Tests if the itemstack is used for citizen interactions
+     *
+     * @param stack
+     * @return
+     */
+    public boolean isInteractionItem(final ItemStack stack)
+    {
+        return ISFOOD.test(stack) || stack.getItem() == Items.BOOK || stack.getItem() == Items.GOLDEN_APPLE || stack.getItem() == Items.CACTUS
+                 || stack.getItem() == Items.GLOWSTONE_DUST;
+    }
+
+    /**
+     * Interaction with children for offering food
+     *
+     * @param usedStack
+     * @param player
+     * @param hand
+     */
+    private void childFoodInteraction(final ItemStack usedStack, final Player player, final InteractionHand hand)
+    {
+        if (usedStack.getDisplayName().getString().toLowerCase(Locale.ENGLISH).contains("cookie"))
+        {
+            usedStack.setCount(usedStack.getCount() - 1);
+            player.setItemInHand(hand, usedStack);
+            interactionCooldown = 100;
+
+            if (!level.isClientSide())
+            {
+                playSound(SoundEvents.GENERIC_EAT, 1.5f, (float) SoundUtils.getRandomPitch(getRandom()));
```

Comment:
Should the child actually gain anything from the cookie, e.g. saturation?

Reply:
oh true I didnt add a gain yet